### DISCUSSION TITLE
Two Body parameters are not embedded if they are separated between dependencies and a route

### DISCUSSION THREAD

**Discussion by KrySeyt** [fastapi/fastapi#10926] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI, Body, Depends

app = FastAPI()


def dependency(b: Annotated[str, Body()],) -> str:
    return b


@app.post(r"/")
def route(
        a: Annotated[str, Body()],
        b: Annotated[str, Depends(dependency)],
) -> str:

    return "success"


def f():
    pass


app.dependency_overrides[f] = lambda: "text"
```

### Description

This route should expect request as 

`
{
  "a": "string",
  "b": "string"
}
`

But it expects just plain text for both values and this doesn't work at all. Just 422

This problem only occurs if any dependency overridden by `app.dependency_overrides`.

If we are using `value: str = Body()` instead of `value: Annotated[str, Body()]` it works well

### Operating System

Linux

### Operating System Details

Linux 6.1.69-1-MANJARO

### FastAPI Version

0.108.0

### Pydantic Version

2.5.3

### Python Version

3.11.6

### Additional Context

Uvicorn version - 0.25.0

**Comment by YuriiMotov** [comment]
This was fixed in FastAPI 0.112.4 (https://fastapi.tiangolo.com/release-notes/#01124)

Test case in details:
<details>

```python
from typing import Annotated

from fastapi import FastAPI, Body, Depends
from fastapi.testclient import TestClient
import pytest

app = FastAPI()


def dependency(b: Annotated[str, Body()]) -> str:
    return b


def another_dep():
    return "c"


@app.post(r"/")
def route(
    a: Annotated[str, Body()],
    b: Annotated[str, Depends(dependency)],
    c: Annotated[str, Depends(another_dep)],
):
    return {"a": a, "b": b, "c": c}



@pytest.mark.parametrize("override", [False, True])
def test_without_dep_override(override: bool):
    if override:
        app.dependency_overrides[another_dep] = lambda: "c"

    client = TestClient(app=app)
    response = client.post(
        "/",
        json={"a": "a", "b": "b", "c": "c"},
    )
    
    if override:
        del app.dependency_overrides[another_dep]

    assert response.status_code == 200, response.json()
    assert response.json() == {"a": "a", "b": "b", "c": "c"}
```

</details>
